packages/util/src/bytes.ts

```diff
@@ -276,3 +276,15 @@ export const baToJSON = function (ba: any): any {
     return array
   }
 }
+
+/**
+ *
+ * @param values An object containing a set keys and RLP encoded values
+ * @returns a string with the value that was encoded with leading zeroes or undefined if no values have leading zeroes
+ */
+export const hasLeadingZeroes = function (values: RlpValues): string | undefined {
+  const res = Object.entries(values).find(
+    (entry) => entry[1] && entry[1]?.length > 0 && entry[1][0] === 0x00
+  )
+  return res ? res[0] : undefined
+}
```

Comment:
Also here, let's please give this a round of discussion if we want expose or not (or rather hang this lower for now and place in `util.ts` tx.

This question is also connected with my following concern: I really wonder if we want to do this check in the Transaction library for the `0x00` value, for example for the nonce. While technically correct I wonder how much hazzle we will produce with this on consuming libraries. It's rather likely that people do something like this in their code bases:

```typescript
import { toBuffer } from 'ethereumjs-util'
const nonce = (SOME_INPUT_NUMBER) // e.g. 0, 1, 2,...

const tx = Transaction.fromValuesArray([ ..., toBuffer(nonce)) ]

```

So this *will* lead to this `0x00` input *even* with our own `toBuffer()` utility method. And at the end this forms a totally valid respectively *the same*  Transaction object as if initialized with `0x`.

So I wonder if this rather has the nature of some "bug fix" or are we rather (eventually unnecessarily) restricting a somewhat more loose user input with more harm than good? 🤔

To circle back, @acolytec3 some base question again: are these new tests actually also triggering on this specific `0x00` case or are they just targeting something like `0x0003`?

So my current tendency is to not go so strict here, but also unsure. Will also circle in the Hardhat guys here 😋 @fvictorio @alcuadrado since they often have a very practical view on these kind of things.

Reply:
>To circle back, @acolytec3 some base question again: are these new tests actually also triggering on this specific 0x00 case or are they just targeting something like 0x0003?

[This test](is an example of one where the nonce passed in is just 0x00 (at least that's what it shows up as when decoded with the `rlp` library.  The set of decoded bytes looks like this:
```
[
  <Buffer 00>,
  <Buffer 01>,
  <Buffer 52 08>,
  <Buffer 09 5e 7b ae a6 a6 c7 c4 c2 df eb 97 7e fa c3 26 af 55 2d 87>,
  <Buffer 0b>,
  <Buffer >,
  <Buffer 1c>,
  <Buffer 48 b5 5b fa 91 5a c7 95 c4 31 97 8d 8a 6a 99 2b 62 8d 55 7d a5 ff 75 9b 30 7d 49 5a 36 64 93 53>,
  <Buffer 10 00 2c ef 53 8b c0 c8 e2 1c 46 08 06 34 a9 3f 4d 75 2b c9 fe 4b 54 6b 60 ac 05 5e 84 2d 34 2b>
]
```

Reply:
Ok, thanks, well, that's the 0 case. 🙂 

Nevertheless let's keep this open for some more time (maybe we need to halt here for 1-2 days).

Side reference, did the following assurance tests on this:

```shell
> rlp.encode('0')
<Buffer 30>
> rlp.encode(Buffer.from('', 'hex'))
<Buffer 80>
> rlp.encode(Buffer.from('00', 'hex'))
<Buffer 00>
> rlp.decode(rlp.encode(Buffer.from('', 'hex')))
<Buffer >
> rlp.decode(rlp.encode(Buffer.from('00', 'hex')))
<Buffer 00>
```

Reply:
@holgerd77 Hmm, I'm not sure I understand this. Can you elaborate on what are the options here and what each one means for consumers?

Reply:
@fvictorio Thanks for chiming in. 🙂 

The options are:

1. The `Tx.fromValuesArray()` function takes as 0-value in the form of a Buffer - for all values which represent integers like e.g. the nonce - both an empty and a zero-Buffer (so both `<Buffer >` and `<Buffer 00>`, as it is now). Theoretically (or: by RLP specification) only the emtpy Buffer would be strictly valid here.
2. We narrow the input here to only the empty Buffer `<Buffer >`

And the question here would be: would 2. be problematic for you guys?

Reply:
Oh, I see. Indeed this is one of those "breaking change of bug fix?" scenarios.

I _think_ we are not doing something like your `[..., toBuffer(nonce)]` example anywhere. The places where we create txs use data that comes from deserializing already valid RLP encoded txs, so this doesn't seem like a problem. And I'm fairly confident that if this causes an issue and you run our test suite with this change, it will be caught.

Also, I personally prefer option 2 if it's more compliant.

Reply:
Ok, thanks' for having a look and the additional dedicated opinion expressed here, that's really valuable. 😄